lib/flowViewport/sceneIndex/fvpSelectionSceneIndex.cpp

```diff
@@ -275,10 +275,9 @@ PrimSelections SelectionSceneIndex::UfePathToPrimSelections(const Ufe::Path& app
             TF_WARN("SelectionSceneIndex::UfePathToPrimSelections(%s) returned no path, Hydra selection will be incorrect", Ufe::PathString::string(appPath).c_str());
         };
 
-        if (!mapper) {
-            warnEmptyPath(appPath);
-        }
-        else {
+        //Don't emit a warning if there is no mapper for this path, _selectionSceneIndex->ReplaceSelection may be called 
+        // later again if there are some usd stage scene index or custom producer scene indices added to the merging scene index.
+        if (mapper) {

```

Comment:
This code is called in initHydraResource so when we switch to a Hydra viewport.
And it may be called later if there are data producer scene indices in the scene and in the second case we have all the scene setup so there should have no errors. But in the first call, the Usd stages prims are not yet in the merging scene index hence the warning.

Reply:
I found a way to keep this warning... Please have a look at the new code.

Reply:
It also calls always once replaceselection instead of potentially twice in my previous change.

Reply:
I see what you meant now, and yeah it is a bit awkward. But I do prefer handling this at the callsites like with the new changes than in the body of this method. Thanks for the changes!